pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -348,9 +348,36 @@ func (a *AWSActuator) getSubnetsByAvailabilityZone(pool *hivev1.MachinePool) (ma
 		return nil, err
 	}
 
-	conflictingSubnets := sets.NewString()
-	subnetsByAvailabilityZone := make(map[string]string, len(pool.Spec.Platform.AWS.Subnets))
+	// Fetch private subnets from the provided list
+	vpc := *results.Subnets[0].VpcId
```

Comment:
hmm.. i don't see a check before that makes sure the slice has atleast one item. we should probably add a check to be sure.

Reply:
This is method (getPrivateSubnetsByAvailabilityZone) is called only when Subnets are specified - https://github.com/openshift/hive/pull/1234/files/4f53e3540d76d08435e5852d52dad266607db9dc#diff-3d3ba26bc232b893654e5bc0fb8bacb63a11a8c559cf415413d9b837c974d8d5L160-L162

Reply:
... On second thought, an additional check wouldn't hurt - especially if kube can't read the subnets  as you described above